src/Microsoft.Azure.WebJobs/ExtensionAttribute.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Description
+{
+    /// <summary>
+    /// Attribute applied to the <see cref="IExtensionConfigProvider"/> implementation for an extension.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public sealed class ExtensionAttribute : Attribute
```

Comment:
Are we going to make this required for ExtensionConfigProviders? (i.e. will tooling require it?)

If it's only meant to be used when the name is different from the class, we could rename this to `ExtensionAlias` to be clear. Logging has a similar `ProviderAlias` attribute.

Reply:
It isn’t required. We’ll fallback to deriving the name from the type based on convention then to use the full type name, I agree that `ExtensionAlias` would be a better option.

Reply:
I named this ExtensionAttribute rather than ExtensionName/ExtensionAlias to allow for the attribute to contain other extension metadata in the future.